OidcDebugger/Views/Home/Index.cshtml

```diff
@@ -1,197 +1,251 @@
+                                    <label for="pkce-sha256" class="select-label">SHA-256</label>
+
+                                </div>
+                                <div class="form-group">
+                                    <input v-on:change="saveParameters" v-model="pkceMethod" type="radio" id="pkce-plain" name="pkce-method" value="plain" />
+                                    <label for="pkce-plain" class="select-label">plain</label>
+
+                                </div>
+                                <div class="form-group">
+                                    <input v-on:change="saveParameters" v-model="pkceMethod" type="radio" id="pkce-disabled" name="pkce-method" value="disabled" />
+                                    <label for="pkce-disabled" class="select-label">disabled</label>
+                                </div>
+                            </div>
+                            <div class="row pt-2">
+                                <div class="col-7">
+                                    <div class="md-form">
+                                        <label for="pkce-code-verifier">Code Verifier</label>
+                                        <input v-on:focus="showInfo" v-on:input="saveParameters" v-model="pkceCodeVerifier" type="text" id="pkce-code-verifier" name="pkce-code-verifier" />
+                                    </div>
+
+                                    <div class="md-form">
+                                        <label for="pkce-code-challenge">Code Challenge</label>
+                                        <input disabled v-model="pkceCodeChallenge" type="text" id="pkce-code-challenge" name="pkce-code-challenge" />
+                                    </div>
+
+                                    <div class="md-form">
+                                        <label for="tokenUri">
+                                            Token URI <span class="required-text">(required for PKCE)</span>
+                                        </label>
+                                        <input type="text" id="tokenUri" class="form-control"
+                                               v-model="tokenUri" v-on:focus="showInfo" v-on:input="saveParameters" required />
+                                    </div>
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+
+            <div class="row">
+                <div class="col-7">
+                    <info-card v-show="flow === 'code'" title="Authorization code flow">
+                        <p>
+                            The authorization server will respond with a <code>code</code>, which the client can exchange for tokens on a secure channel.
+                            This flow should be used when the application code runs on a secure server (common for MVC and server-rendered pages apps).
```

Comment:
If PKCE is enabled, hide this second paragraph and replace it with:

> With PKCE, the browser can act as a secure channel and perform the token exchange. The tokens will be returned directly to the browser.

@bdemers Do you like this language?

Reply:
I'm going to kick this to @aaronpk (specifically cause I'm stealing bits from [oauth.net](https://oauth.net/2/grant-types/authorization-code/)):

```html
The Authorization Code grant type is used by confidential and public clients to exchange an authorization <code>code</code> for an access token.
```

but maybe we change it to something more generic like:

```html
The authorization server will respond with a <code>code</code>, which the client can exchange for tokens.
It is recommended that all clients use PKCE extension with this flow as well to provide better security.
```